packages/core/src/hydration/i18n.ts

```diff
@@ -154,39 +186,98 @@ export function trySerializeI18nBlock(
     return null;
   }
 
-  const caseQueue: number[] = [];
-  tI18n.ast.forEach((node) => serializeI18nBlock(lView, caseQueue, context, node));
-  return caseQueue.length > 0 ? caseQueue : null;
+  const serializedI18nBlock: SerializedI18nBlock = {
+    caseQueue: [],
+    disconnectedNodes: new Set(),
+    disjointNodes: new Set(),
+  };
+  serializeI18nBlock(lView, serializedI18nBlock, context, tI18n.ast, false /* isICUChild */);
+
+  return serializedI18nBlock.caseQueue.length === 0 &&
+    serializedI18nBlock.disconnectedNodes.size === 0 &&
+    serializedI18nBlock.disjointNodes.size === 0
+    ? null
+    : serializedI18nBlock;
 }
 
 function serializeI18nBlock(
   lView: LView,
-  caseQueue: number[],
+  serializedI18nBlock: SerializedI18nBlock,
+  context: HydrationContext,
+  nodes: I18nNode[],
+  isICUChild: boolean,
+): Node | null {
+  let prevRNode = null;
+  for (const node of nodes) {
+    const nextRNode = serializeI18nNode(lView, serializedI18nBlock, context, node, isICUChild);
+    if (nextRNode && !isICUChild) {
+      if (prevRNode && prevRNode.nextSibling !== nextRNode) {
```

Comment:
nit: we can extract this logic into a separate function and move the comment (below) as a function description:
```suggestion
      if (isDisjointNode(prevRNode, nextRNode)) {
```